packages/suite/src/components/wallet/TokenIconSetWrapper.tsx

```diff
@@ -3,30 +3,57 @@ import {
     enhanceTokensWithRates,
     getTokens,
     sortTokensWithRates,
+    TokensWithRates,
 } from 'src/utils/wallet/tokenUtils';
 import { TokenIconSet } from '@trezor/product-components';
 import { useSelector } from 'src/hooks/suite';
 import { selectLocalCurrency } from 'src/reducers/wallet/settingsReducer';
 import { selectCurrentFiatRates } from '@suite-common/wallet-core';
 import { Account } from '@suite-common/wallet-types';
+import { NetworkSymbol } from '@suite-common/wallet-config';
 
 type TokenIconSetWrapperProps = {
-    account: Account;
+    accounts: Account[];
+    network: NetworkSymbol;
 };
 
-export const TokenIconSetWrapper = ({ account }: TokenIconSetWrapperProps) => {
-    const { tokens: accountTokens, symbol: network } = account;
+export const TokenIconSetWrapper = ({ accounts, network }: TokenIconSetWrapperProps) => {
+    const localCurrency = useSelector(selectLocalCurrency);
     const fiatRates = useSelector(selectCurrentFiatRates);
     const coinDefinitions = useSelector(state => selectCoinDefinitions(state, network));
-    const localCurrency = useSelector(selectLocalCurrency);
-    const tokensWithRates = enhanceTokensWithRates(
-        accountTokens,
-        localCurrency,
-        network,
-        fiatRates,
+
+    const allTokensWithRates = accounts.flatMap(account =>
+        enhanceTokensWithRates(account.tokens, localCurrency, network, fiatRates),
     );
-    const sortedTokens = tokensWithRates.sort(sortTokensWithRates);
-    const tokens = getTokens(sortedTokens, network, coinDefinitions);
 
-    return <TokenIconSet network={network} tokens={tokens.shownWithBalance} />;
+    if (!allTokensWithRates.length) return null;
+
+    const tokens = getTokens(allTokensWithRates, network, coinDefinitions)
+        .shownWithBalance as TokensWithRates[];
+
+    const aggregatedTokens = Object.values(
+        tokens.reduce((acc: Record<string, TokensWithRates>, token) => {
+            const { contract, balance, fiatValue } = token;
+
+            if (!acc[contract]) {
+                acc[contract] = {
+                    ...token,
+                    balance: balance ?? '0',
+                    fiatValue: fiatValue ?? { s: 0, e: 0, c: [0] },
```

Comment:
what is s, e, c?

Reply:
It is a BigNumber as an object, but probably forgot to change it to a normal BigNumber, here it is fixed https://github.com/trezor/trezor-suite/pull/14996/commits/1a6ffd52c4948eb5c17e79078aaab4088ffadd44